carapax-i18n/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+#[cfg(feature = "gettext")]
+mod gettext;
+#[cfg(feature = "gettext")]
+pub use gettext::Gettext;
+
+use carapax::{core::types::Message, Context, HandlerFuture, HandlerResult, MessageHandler};
+use std::collections::HashMap;
+
+/// An i18n handler for carapax
+pub struct I18nHandler<T> {
+    default: T,
+    languages: HashMap<String, T>,
+}
+
+impl<T> I18nHandler<T>
+where
+    T: Translator,
+{
+    /// Creates new I18nHandler
+    ///
+    /// Search translator in languages or use default one
+    pub fn new(default: T, languages: HashMap<String, T>) -> Self {
+        Self { default, languages }
+    }
+}
+
+impl<T> MessageHandler for I18nHandler<T>
+where
+    T: Translator + Send + Sync + 'static,
+{
+    fn handle(&self, context: &mut Context, message: &Message) -> HandlerFuture {
+        let translator = message
+            .get_user()
+            .and_then(|user| {
+                user.language_code.clone().and_then(|lang| {
```

Comment:
What if I need to change language in a bot settings?
I guess we need to introduce a TranslatorResolver trait.

Reply:
this trait should have a method which returns a language code based on update

Reply:
Not message, update, so you need to implement UpdateHandler instead of MessageHandler